### PR TITLE
Setup WebSocket server.

### PR DESCRIPTION
**Pull request by duyoji**
- This PR is for #36.
- When we need to modify to connect to client, I will create new PR for that.
- Formatting response data by socket will be implemented.
  - Just created frame of Socket.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -28,7 +28,10 @@
     "react": "^16.2.0",
     "react-dom": "^16.2.0",
     "react-scripts": "^1.0.17",
-    "redis": "^2.8.0"
+    "redis": "^2.8.0",
+    "socket.io": "^2.0.4",
+    "socket.io-redis": "^5.2.0",
+    "sticky-session": "^1.1.2"
```
Comment (duyoji) [inline package.json]: https://qiita.com/takehilo/items/8c773d18ec6cfccd6679

Referred to this article. (Japanese article.)

File: server/socket.js [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+  const isWorker = sticky.listen(server, port);
+  console.log(`localhost:${port}`);
+
+  if (isWorker) {
+    const game = io.of('/game').on('connection', (socket) => {
+      console.log(`connected, id: ${socket.id}`);
+
+      socket.on('getRooms', () => {
+        game.adapter.allRooms((err, rooms) => {
+          console.log(rooms); // an array containing all rooms (accross every node)
```
Comment (duyoji) [inline server/socket.js]: When I start to write more in detail, I will remove console.log in this file. (This PR is for just settingup env.)

**Comment by duyoji** [comment]
@ryutaudo 

Could you review this PR if you have time?

File: server/index.js [hunk 3]
```diff
@@ -1,11 +1,6 @@
...
 const PORT = process.env.PORT || 9000;
 
-app.listen(PORT, () => {
-  console.log(
-    chalk.green.bold('Server listening on port: ')
-    + chalk.cyan.bold(PORT)
-  )
-});
\ No newline at end of file
+listen(app, PORT);
```
Comment (therealmeowzedong) [inline server/index.js]: 👍

File: server/socket.js [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+const listen = (expressApp, port = DEFAULT_PORT) => {
+  const io = socketIO();
+  const server = http.createServer(expressApp);
+
+  io.adapter(redisAdapter(process.env.REDIS_URL));
+  io.attach(server);
+  const isWorker = sticky.listen(server, port);
+
+  if (isWorker) {
+    const game = io.of('/game').on('connection', (socket) => {
```
Comment (therealmeowzedong) [inline server/socket.js]: This block is rather large...

Comment (duyoji) [reply server/socket.js]: I created files for each event handler in `server/socketHandlers` directory. But now I don't have test code for that. From now I will create Test code for those.

**Comment by duyoji** [comment]
Additional task of `websocket` is worked on #47

File: server/socket.js (hunk 4 above)
Comment (duyoji) [reply server/socket.js]: Created tests for each socketHandler in `server/__tests__/socketHandlers` directory.

**Comment by duyoji** [comment]
@ryutaudo @therealmeowzedong 

I fixed code based on review.
Please review this PR when you have spare time.

**Comment by kumiko-haraguchi** [comment]
@duyoji a little comment, run `yarn eslint` :)

**Comment by duyoji** [comment]
> @duyoji a little comment, run yarn eslint :)

I fixed lint errors of backend code.

https://github.com/duyoji/oni-on-cheese/pull/41/commits/e9112318af44fafb9f3809c74f2a8d08537c67e7

**Comment by duyoji** [comment]
According to below comment, I will merge this PR.

image

**Pull request merged**